### PR TITLE
Improve tray icon behavior

### PR DESCRIPTION
**Pull request by bemyak**
**What's this PR do?**
This PR improves behaviour of task bar icon on all OSes.
The main logic is simple:
- If you click on task bar icon and you don't see zulip's window, it is getting a focus.
- If you click on task bar icon and you see a window, it is getting hidden

**Any background context you want to provide?**
The current version handles clicks only on windows and completely ignores it on other OSes

**You have tested this PR on:**
  - [ ] Windows
  - [X] Linux/Ubuntu
  - [ ] macOS

### CODE REVIEW

File: app/renderer/js/tray.js [hunk 1]
```diff
@@ -154,10 +154,7 @@ const createTray = function () {
 	]);
 	window.tray.setContextMenu(contextMenu);
 	window.tray.on('click', () => {
-		// Click event only works on Windows
-		if (process.platform === 'win32') {
-			ipcRenderer.send('toggle-app');
-		}
+		ipcRenderer.send('toggle-app');
```
Comment (abhigyank) [inline app/renderer/js/tray.js]: Is this now supported across all OS?

Comment (bemyak) [reply app/renderer/js/tray.js]: I tested it on Linux. Windows was supported already. I haven't got access to MacOs, but I saw other other apps (ex. bitwarden) use the same method for all OSes

**Comment by akashnimare** [comment]
@bemyak I don't think the click event is supported on macOS. Just tested this and there was no effect.

**Comment by bemyak** [comment]
Should I check if `process.platform !== 'darwin'`? Or I can leave it as it is and hope that support will be added some day?

**Comment by bemyak** [comment]
I've update this PR to disable click event subscription on MacOs. I've also found a friend of mine with mac. He says that the tray icon works as expected, just like other icons.

**Comment by akashnimare** [comment]
> I've also found a friend of mine with mac. He says that the tray icon works as expected, just like other icons.

Can you post a screenshot if possible? For me, the click event is never called on macOS and Linux. @abhigyank told me that it works on his Ubuntu system. @bemyak are you using the unity? Can you also post a gif of this?

**Comment by bemyak** [comment]
Yes, I'll post a gif later. I use KDE and the tray icon was not working at all there. To bring the app up I needed to right-click on it, and select "Focus".

**Comment by bemyak** [comment]
Finally managed to capture the screen =)
https://youtu.be/W3bnZKsty_w
In the beginning I'm on current HEAD of zulip repo, later I switched to my fork.

**Comment by bemyak** [comment]
Can I do something else to get this merged?

File: app/main/index.js [hunk 2]
```diff
@@ -209,10 +209,10 @@ app.on('ready', () => {
 	});
 
 	ipcMain.on('toggle-app', () => {
-		if (mainWindow.isVisible()) {
-			mainWindow.hide();
-		} else {
+		if (!mainWindow.isVisible() || mainWindow.isMinimized() || !mainWindow.isFocused()) {
```
Comment (akashnimare) [inline app/main/index.js]: @bemyak this doesn't work for me on Windows, can you keep the old method?

**Review by akashnimare — CHANGES_REQUESTED** [review]

File: app/main/index.js (hunk 2 above)
Comment (bemyak) [reply app/main/index.js]: Thank you!
I've tested it on Windows today. The problem was in `isFocused()` part of condition.
The reason is simple — when you click on tray on Windows the tray gets the focus for a second and app's window gets unfocused.
I remove the condition and now everything works as expected.

**Comment by bemyak** [comment]
Hi! Can I do anything else on this issue?

**Comment by akashnimare** [comment]
@bemyak sorry, I forgot to comment. The toggling feature stopped working for me on Windows when I checked this PR. Can you test it again?

**Comment by bemyak** [comment]
I've tested it just now, works fine:
https://youtu.be/xp5tPcpcEMM

**Review by akashnimare — APPROVED** [review]

**Comment by akashnimare** [comment]
Merged. Thanks, @bemyak (sorry for the delay, was busy in other stuff).

**Pull request merged**